Import package pandas 

In [1]:
import pandas as pd

Inputkan data dari http://bit.ly/drinksbycountry. Data berikut merupakan data konsumsi minuman dengan satuan  litres per capita of drinking-age population.

In [2]:
df = pd.read_csv("http://bit.ly/drinksbycountry")
df.head()

country  beer_servings  spirit_servings  wine_servings  \
0  Afghanistan              0                0              0   
1      Albania             89              132             54   
2      Algeria             25                0             14   
3      Andorra            245              138            312   
4       Angola            217               57             45   

   total_litres_of_pure_alcohol continent  
0                           0.0      Asia  
1                           4.9    Europe  
2                           0.7    Africa  
3                          12.4    Europe  
4                           5.9    Africa

Data dari informasi dari tabel diatas mempunyai 6 variabel yaitu country (liter), beer servings (liter), spirit servings (liter), wine servings (liter), total litres of pure alcohol dan continent. 

Melihat info data

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 193 entries, 0 to 192
Data columns (total 6 columns):
country                         193 non-null object
beer_servings                   193 non-null int64
spirit_servings                 193 non-null int64
wine_servings                   193 non-null int64
total_litres_of_pure_alcohol    193 non-null float64
continent                       193 non-null object
dtypes: float64(1), int64(3), object(2)
memory usage: 9.1+ KB


Dari info data diatas terdapat baris data sebanyak 193 baris data. Dengan 193 baris data untuk semua variabel baik country, beer servings, spirit servings, wine servings, total litres of pure alcohol dan continent. Oleh karena itu, diartikan bahwa nilai pada semua variabel terisi atau tidak ada data yang kosong. 

Melihat banyaknya negara setiap benua (continent) 

In [4]:
df['continent'].value_counts()

Africa           53
Europe           45
Asia             44
North America    23
Oceania          16
South America    12
Name: continent, dtype: int64

Dari data diatas menyampaikan banyak baris data (banyaknya negara) sesuai dengan Benua (continent) yaitu Afrika sebanyak 53 negara, Europe sebanyak 45 negara, Asia sebanyak 44 negara dan seterusnya dapat dilihat pada output diatas. 

Melihat deskripsi statistika data untuk semua data

In [7]:
df.describe()

beer_servings  spirit_servings  wine_servings  \
count     193.000000       193.000000     193.000000   
mean      106.160622        80.994819      49.450777   
std       101.143103        88.284312      79.697598   
min         0.000000         0.000000       0.000000   
25%        20.000000         4.000000       1.000000   
50%        76.000000        56.000000       8.000000   
75%       188.000000       128.000000      59.000000   
max       376.000000       438.000000     370.000000   

       total_litres_of_pure_alcohol  
count                    193.000000  
mean                       4.717098  
std                        3.773298  
min                        0.000000  
25%                        1.300000  
50%                        4.200000  
75%                        7.200000  
max                       14.400000

Dari tabel diatas menghitung analisa statistika deskriptif. Pada beer servings mempunyai rata-rata 106.16 liter dari semua negara, Standar deviasi sebesar 101.143 liter, Nilai minimal atau nilai terkecil sebesar 0 liter, Quartil 1 (25%) sebesar 20 liter, Quartil 2 (50%) sebesar 76 liter, Quartil 3 (75%) sebesar 188 liter dan nilai maximal atau nilai tertinggi yaitu sebesar 376 liter, dan seterusnnya untuk variabel lain dapat dilihat pada tabel output diatas.  

Mengurutkan data sesuai dengan variabel Benua (continent) dan urutkan sesuai abjad, tujuannya agar melihat data sesuai dengan kelompok Benua dengan mudah. 

In [10]:
df1 = df.sort_values('continent').reset_index(drop=True)
df1.head()

country  beer_servings  spirit_servings  wine_servings  \
0     Liberia             19              152              2   
1        Mali              5                1              1   
2      Malawi              8               11              1   
3  Madagascar             26               15              4   
4       Libya              0                0              0   

   total_litres_of_pure_alcohol continent  
0                           3.1    Africa  
1                           0.6    Africa  
2                           1.5    Africa  
3                           0.8    Africa  
4                           0.0    Africa

Misalnya, kita ingin mengetahui data dari negara Indonesia 

In [13]:
df[df['country']=="Indonesia"]

country  beer_servings  spirit_servings  wine_servings  \
78  Indonesia              5                1              0   

    total_litres_of_pure_alcohol continent  
78                           0.1      Asia

Diketahui bahwa data dari negara Indonesia dengan beer servings sebesar 5 liter, spirit servings sebesar 1 liter, wine servings sebesar 0 liter dan total litres of pure alcohol sebesar 0.1 liter. Disimpulkan dari rata-rata pernegara negera Indonesia mempunyai minuman yang sangat kecil. 

Total semua konsumsi dari semua negara pada data set.

In [39]:
df.sum(numeric_only=True)

beer_servings                   20489.0
spirit_servings                 15632.0
wine_servings                    9544.0
total_litres_of_pure_alcohol      910.4
dtype: float64

beer servings mepunyai total konsumsi dari semua negara untuk beer servings sebesar 20489 liter, spirit servings sebesar 15632 liter, wine servings sebesar 9544 dan total litres of pure alcohol sebesar 910 liter. 

Total konsumsi minuman berdasarkan Benua (continent)

In [15]:
df.groupby('continent', as_index=False).sum()

continent  beer_servings  spirit_servings  wine_servings  \
0         Africa           3258              866            862   
1           Asia           1630             2677            399   
2         Europe           8720             5965           6400   
3  North America           3345             3812            564   
4        Oceania           1435              935            570   
5  South America           2101             1377            749   

   total_litres_of_pure_alcohol  
0                         159.4  
1                          95.5  
2                         387.8  
3                         137.9  
4                          54.1  
5                          75.7

Pada benua Afrika total konsumsi minuman beer servings sebesar 3158 liter, Benua Asia 1630 liter, Benua Eropa 8720  liter, dan seterusnya dapat dilihat pada tabel diatas. Didapatkan konsumsi beer survings paling tinggi yaitu Benua Eropa dengan 8720 liter dan Benua dengan konsumsi terendah Oceania dengan 1435 liter. Dan lainnya dapat dilihat pada tabel diatas. 

Melihat nilai konsumsi tertinggi pada setiap benua. 

In [49]:
df4 = df.groupby(['continent'], as_index=False).max()
df4.drop(columns='country')

continent  beer_servings  spirit_servings  wine_servings  \
0         Africa            376              152            233   
1           Asia            247              326            123   
2         Europe            361              373            370   
3  North America            285              438            100   
4        Oceania            306              254            212   
5  South America            333              302            221   

   total_litres_of_pure_alcohol  
0                           9.1  
1                          11.5  
2                          14.4  
3                          11.9  
4                          10.4  
5                           8.3

Dari tabel diatas konsumsi tertinggi pada Benua Afrika untuk minuman beer servings sebesar 376 liter, sirit serving sebesar 152 liter, wine serving sebesar 233 liter dan total litres of pure alcohol sebesar 9.1 liter. Dan Benua lainnya dapat dilihat pada tabel diatas. 

Rata-rata konsumsi minuman setiap negara berdasarkan benua. 

In [46]:
df.groupby(['continent'], as_index=False).mean()

continent  beer_servings  spirit_servings  wine_servings  \
0         Africa      61.471698        16.339623      16.264151   
1           Asia      37.045455        60.840909       9.068182   
2         Europe     193.777778       132.555556     142.222222   
3  North America     145.434783       165.739130      24.521739   
4        Oceania      89.687500        58.437500      35.625000   
5  South America     175.083333       114.750000      62.416667   

   total_litres_of_pure_alcohol  
0                      3.007547  
1                      2.170455  
2                      8.617778  
3                      5.995652  
4                      3.381250  
5                      6.308333

Rata-rata konsumsi minuman beer serving pada benua Afrika yaitu sebesar 61.47 liter, spirit serving dengan rata-rata 16.34 liter, wine serving dengan rata-rata 16.26 liter dan total litres of pure alcohol sebesar 3.01 liter. 

Menampilkan konsumsi terendah dari setiap benua

In [50]:
df5 = df.groupby(['continent'], as_index=False).min()
df5.drop(columns='country')

continent  beer_servings  spirit_servings  wine_servings  \
0         Africa              0                0              0   
1           Asia              0                0              0   
2         Europe              0                0              0   
3  North America              1               68              1   
4        Oceania              0                0              0   
5  South America             93               25              1   

   total_litres_of_pure_alcohol  
0                           0.0  
1                           0.0  
2                           0.0  
3                           2.2  
4                           0.0  
5                           3.8

Pada benua afrika konsumsi terendah semua minuman yaitu 0 liter. Benua lainnya dapat dilihat pada tabel diatas. 

Membandingkan konsumsi minuman indonesia dan Malaysia

In [99]:
df[(df['country']=="Indonesia") | (df['country']=="Malaysia")]

country  beer_servings  spirit_servings  wine_servings  \
78   Indonesia              5                1              0   
102   Malaysia             13                4              0   

     total_litres_of_pure_alcohol continent  
78                            0.1      Asia  
102                           0.3      Asia

syntax diatas dapat digunakan lagi dengan cara berikut

In [104]:
negara = ['Indonesia', 'Malaysia']
df[df['country'].isin(negara)]

country  beer_servings  spirit_servings  wine_servings  \
78   Indonesia              5                1              0   
102   Malaysia             13                4              0   

     total_litres_of_pure_alcohol continent  
78                            0.1      Asia  
102                           0.3      Asia

Pada tabel diatas diketahui tingkat konsumsi minuman pada negara Malaysia lebih tinggi dari pada negara Indonesia yaitu minuman beer, spirit dan total litres of pure alcohol. Sedangkan minuman wine sama yaitu 0 liter. 

Melihat data dibenua Asia

In [105]:
df2 = df[df['continent'] == 'Asia'].reset_index(drop=True)
df2

country  beer_servings  spirit_servings  wine_servings  \
0            Afghanistan              0                0              0   
1                Bahrain             42               63              7   
2             Bangladesh              0                0              0   
3                 Bhutan             23                0              0   
4                 Brunei             31                2              1   
5               Cambodia             57               65              1   
6                  China             79              192              8   
7            North Korea              0                0              0   
8                  India              9              114              0   
9              Indonesia              5                1              0   
10                  Iran              0                0              0   
11                  Iraq              9                3              0   
12                Israel             63               69              9   
13                 Japan             77              202             16   
14                Jordan              6               21              1   
15            Kazakhstan            124              246             12   
16                Kuwait              0                0              0   
17            Kyrgyzstan             31               97              6   
18                  Laos             62                0            123   
19               Lebanon             20               55             31   
20              Malaysia             13                4              0   
21              Maldives              0                0              0   
22              Mongolia             77              189              8   
23               Myanmar              5                1              0   
24                 Nepal              5                6              0   
25                  Oman             22               16              1   
26              Pakistan              0                0              0   
27           Philippines             71              186              1   
28                 Qatar              1               42              7   
29           South Korea            140               16              9   
30    Russian Federation            247              326             73   
31          Saudi Arabia              0                5              0   
32             Singapore             60               12             11   
33             Sri Lanka             16              104              0   
34                 Syria              5               35             16   
35            Tajikistan              2               15              0   
36              Thailand             99              258              1   
37           Timor-Leste              1                1              4   
38                Turkey             51               22              7   
39          Turkmenistan             19               71             32   
40  United Arab Emirates             16              135              5   
41            Uzbekistan             25              101              8   
42               Vietnam            111                2              1   
43                 Yemen              6                0              0   

    total_litres_of_pure_alcohol continent  
0                            0.0      Asia  
1                            2.0      Asia  
2                            0.0      Asia  
3                            0.4      Asia  
4                            0.6      Asia  
5                            2.2      Asia  
6                            5.0      Asia  
7                            0.0      Asia  
8                            2.2      Asia  
9                            0.1      Asia  
10                           0.0      Asia  
11                           0.2      Asia  
12                           2.5      Asia  
13         

Menampilkan data pada benua Asia dengan variabel country dan beer saja. 

In [106]:
df3 = df2[['country','beer_servings']]

Melihat konsumsi beer terendah di Asia

In [107]:
totals = df3.sort_values('beer_servings', 
                           ascending=False).reset_index(drop=True)
totals.tail(8)

country  beer_servings
36      Pakistan              0
37  Saudi Arabia              0
38      Maldives              0
39        Kuwait              0
40          Iran              0
41   North Korea              0
42    Bangladesh              0
43   Afghanistan              0

atau

In [111]:
df3[df3['beer_servings'] == 0].reset_index(drop=True)

country  beer_servings
0   Afghanistan              0
1    Bangladesh              0
2   North Korea              0
3          Iran              0
4        Kuwait              0
5      Maldives              0
6      Pakistan              0
7  Saudi Arabia              0

Konsumsi minuman terendah yaitu Pakistan, Saudi Arabia, Maldives, Kuwait, Iran, Korea Utara, Banglades dan Afganistan dengan konsumsi beer 0 liter. 

Melihat konsumsi beer tertinggi 

In [108]:
totals = df3.sort_values('beer_servings', 
                           ascending=False).reset_index(drop=True)
totals.head(1)

country  beer_servings
0  Russian Federation            247

Konsumsi tertinggi pada benua Asia yaitu negara Russian Federation sebesat 247 liter. 

https://www.ritchieng.com/pandas-pandas-index/